### ISSUE TITLE
[Bug] Items of FlexLayout overlap with the controls below it

### ISSUE THREAD

**Issue by CodingAthlete** [xamarin/Xamarin.Forms#12793]
### Description

I have created a FlexLayout (Wrap="Wrap") within the left column of a Grid with two columns. The Grid is within a StackLayout. Below the Grid are more controls within the StackLayout. When there is a second row in the FlexLayout and there is one element in this second row, the height of the FlexLayout is too small. So the controls of the FlexLayout overlap with the controls below it.

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="..."
             Title="FlexLayout">
    <ContentPage.Content>
        <StackLayout>
            <Grid>
                <Grid.RowDefinitions>
                    <RowDefinition Height="Auto" />
                    <RowDefinition Height="Auto" />
                </Grid.RowDefinitions>
                <Grid.ColumnDefinitions>
                    <ColumnDefinition Width="*" />
                    <ColumnDefinition Width="Auto" />
                </Grid.ColumnDefinitions>
                <FlexLayout Grid.Row="0" Grid.Column="0" Wrap="Wrap">
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                    <Button FlexLayout.AlignSelf="Stretch" Text="×" HeightRequest="35.0" WidthRequest="50.0" Margin="0.0, 0.0, 1.0, 0.0" BackgroundColor="DarkRed" />
                </FlexLayout>
                <Button Grid.Row="0" Grid.Column="1" HeightRequest="35.0" WidthRequest="50.0" Text="R" />
            </Grid>
            <Label Text="Text below FlexLayout" />
        </StackLayout>
    </ContentPage.Content>
</ContentPage>
```

### Steps to Reproduce

1. execute the attached code
2. change the screen size so the FlexLayout has two rows and the second row has one element

### Expected Behavior

The controls do not overlap and the text is below the FlexLayout.

### Actual Behavior

The height of the FlexLayout is too small and so its elements overlap with the controls below.

### Basic Information

- Version with issue: Xamarin.Forms 4.8.0.1560
- Last known good version: ?
- IDE: Visual Studio 2019
- Platform Target Frameworks:
  - iOS: never tested
  - Android: v9.0
  - UWP:  TargetPlatformVersion: 10.0.18362.0, TargetPlatformMinVersion: 10.0.16299.0
- Android Support Library Version: ?
- Nuget Packages: NETStandard.Library 2.0.3; Xamarin.Forms 4.8.0.1560
- Affected Devices: Desktop Windows 10, HUAWEI Mate 20 lite SNE-LX1 with Android 10

### Screenshots

UWP (Desktop Windows 10):
![FlexLayout](https://user-images.githubusercontent.com/9087189/98644683-2705e700-2331-11eb-9b9e-1f900ab507df.jpg)

**Comment by GUOLDEV** [comment]
Hi all, any update about this issue? Any workaround? For now I am wrapping the FlexLayout with a StackLayout, it works but I would like to remove it once it is fixed.

**Comment by CodingAthlete** [comment]
I tried it again with a newer version of Xamarin.Forms and it seems to fixed since version 4.8.0.1687.
